### PR TITLE
Fix problems mapping with python 3.10

### PR DESCRIPTION
**Pull request by apotterri**
Fixes getappmap/board#150.

### CODE REVIEW

File: appmap/_implementation/recording.py [hunk 1]
```diff
@@ -355,7 +355,18 @@ def wrapped_find_spec(find_spec, _, args, kwargs):
     if spec is not None:
         if getattr(spec.loader, "exec_module", None) is not None:
             loader = spec.loader
-            loader.exec_module = wrap_exec_module(loader.exec_module)
+            # This is kind of gross. As this comment linked to below describes, wrapt has trouble
+            # identifying methods decorated with @staticmethod. It offers two suggested fixes:
+            # update the class definition, or patch the function in __dict__. We can't do the
+            # former, so do the latter instead.
+            #   comment in wrapt: https://github.com/GrahamDumpleton/wrapt/blob/master/src/wrapt/wrappers.py#L730
```
Comment (dividedmind) [inline appmap/_implementation/recording.py:362]: This should probably use a permalink with commit id. You can easily get one by pressing the 'Y' key in github.

**Review by dividedmind — APPROVED** [review]
Looks good. Had one comment (on a comment, on a comment).

BTW, I wonder if it wouldn't be a good idea to take a good look at the descriptor protocol and see if we can roll our own code to improve on what wrapt is doing for us. It sometimes seems it has almost as many corner cases we have to handle as we'd have to when doing it directly.

File: .travis.yml [hunk 2]
```diff
@@ -11,17 +14,20 @@ env:
...
+  eval "$(pyenv init -)"
+  eval "$(pyenv init --path)"
+  eval "$(pyenv virtualenv-init -)"
   pyenv update
-  git clone https://github.com/momo-lab/xxenv-latest.git "$(pyenv root)/plugins/xxenv-latest"
-  for v in {7..9}; do pyenv latest install -s 3.$v; pyenv latest uninstall -f 3.$v; done
-  pyenv latest local 3.{7..9}
-  pip install --upgrade pip
-  pip install poetry tox
+  for v in {7..10}; do pyenv latest install -s 3.$v; pyenv latest uninstall -f 3.$v; done
```
Comment (dividedmind) [inline .travis.yml:28]: I wonder, is there a distro or another way in travis to have all these pythons precompiled? If not, maybe we could pull binaries? This really seems a huge waste of cycles.

File: appmap/_implementation/recording.py [hunk 3]
```diff
@@ -355,7 +355,18 @@ def wrapped_find_spec(find_spec, _, args, kwargs):
...
+            #
+            # TODO: determine if we can use wrapt.wrap_function_wrapper to simplify this code
+            exec_module = inspect.getattr_static(loader, "exec_module")
+            if isinstance(exec_module, staticmethod):
+                loader.exec_module = wrap_exec_module(exec_module)
```
Comment (dividedmind) [inline appmap/_implementation/recording.py:367]: I don't entirely understand how it's different. Surely if `exec_module` is a `staticmethod`, `loader.exec_module === exec_module`? Or is there something else going on?

Comment (dividedmind) [reply appmap/_implementation/recording.py:367]: Ah, never mind, actually `loader.exec_module == exec_module.__wrapped__` (or whatever is the attr called).

File: appmap/_implementation/recording.py (hunk 1 above)
Comment (apotterri) [reply appmap/_implementation/recording.py:362]: Yes, you're right. GH even did the right thing when I copied the link, and I changed it `master`. Oops. I'll change it back.

File: .travis.yml (hunk 2 above)
Comment (apotterri) [reply .travis.yml:28]: The config is supposed to be telling Travis to cache the `pyenv` installs:
```
cache:
- $PYENV_ROOT
```
and this
```
for v in {7..10}; do pyenv latest install -s 3.$v; pyenv latest uninstall -f 3.$v; done
```
is supposed to be ensuring that just the latest release is in the cache: https://github.com/momo-lab/xxenv-latest#usage .

After I get this PR merged, I'll investigate why this isn't working, and improve it (somehow).

File: appmap/_implementation/recording.py (hunk 3 above)
Comment (apotterri) [reply appmap/_implementation/recording.py:367]: Yeah:
```
ajp@Alans-MacBook-Pro appmap-python % python
Python 3.10.7 (main, Sep 15 2022, 04:53:36) [Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from importlib.machinery import BuiltinImporter
>>> import inspect
>>> getattr(BuiltinImporter, 'exec_module')
<function BuiltinImporter.exec_module at 0x10edde4d0>
>>> inspect.getattr_static(BuiltinImporter, 'exec_module')
<staticmethod(<function BuiltinImporter.exec_module at 0x10edde4d0>)>
```

File: .travis.yml (hunk 2 above)
Comment (apotterri) [reply .travis.yml:28]: FWIW, builds on the same branch do cache $PYENV_ROOT correctly:
https://app.travis-ci.com/github/getappmap/appmap-python/jobs/584499355#L262 .

The build for the merged PR didn't use the cache though. :( So, there's definitely room for improvement here.

**Comment by apotterri** [comment]
> Looks good. Had one comment (on a comment, on a comment).
> 
> BTW, I wonder if it wouldn't be a good idea to take a good look at the descriptor protocol and see if we can roll our own code to improve on what wrapt is doing for us. It sometimes seems it has almost as many corner cases we have to handle as we'd have to when doing it directly.

Yeah, I agree that we might be better off without `wrapt`. Coming up with a different strategy is on my list of things to do post Tech Crunch.

File: .travis.yml (hunk 2 above)
Comment (dividedmind) [reply .travis.yml:28]: Thanks for looking into this!

**Pull request merged**